rhythm.py

```diff
@@ -1,24 +1,81 @@
+import logging
+from abc import ABCMeta, abstractmethod
+from time import sleep
+
 from regression import calculate_regression
-import math
 
-HANDSTROKE_GAP = 1
 
-class Rhythm:
-    logger_name = "RHYTHM"
+class Rhythm(metaclass=ABCMeta):
+    @abstractmethod
+    def wait_for_bell_time(self, current_time, bell, row_number, place, user_controlled, stroke):
+        pass
+
+    @abstractmethod
+    def expect_bell(self, expected_bell, row_number, index, expected_stroke):
+        pass
+
+    @abstractmethod
+    def on_bell_ring(self, bell, stroke, real_time):
+        pass
+
+    @abstractmethod
+    def initialise_line(self, stage, user_controls_treble, start_time):
+        pass
+
+
+class WaitForUserRhythm(Rhythm):
+    logger_name = "RHYTHM:WaitForUser"
+
+    def __init__(self, rhythm: Rhythm):
+        self._innerRhythm = rhythm
+        self._expected_bells = set()
+        self.delay = 0
+        self.logger = logging.getLogger(self.logger_name)
+
+    def wait_for_bell_time(self, current_time, bell, row_number, place, user_controlled, stroke):
+        self._innerRhythm.wait_for_bell_time(current_time - self.delay, bell, row_number, place, user_controlled, stroke)
```

Comment:
Being able to pass times to the inner rhythm having adjusted the time passed in is a neat result of not you not making `Rhythm` itself dependent on `time.time()` ❤️